makeflow/src/makeflow.c

```diff
@@ -1523,6 +1538,14 @@ int main(int argc, char *argv[])
 				if(!wrapper) wrapper = makeflow_wrapper_create();
 				makeflow_wrapper_add_output_file(wrapper, optarg);
 				break;
+			case LONG_OPT_SHARED_FS:
+				assert(shared_fs);
+				s = string_combine(xxstrdup(optarg), "/");
+				t = realpath(s, NULL);
+				free(s);
+				if (!t) fatal("can't access shared filesystem %s: %s", optarg, strerror(errno));
+				list_push_head(shared_fs, string_combine(t, "/"));
```

Comment:
IMHO, it's best not to "massage" file names unless there is a really compelling reason, otherwise confusing things can happen.  It's probably simplest to just store the string given by the user, and compare it to the string names in the batch jobs.  (The individual files will have their existence checked anyhow.)